lisa/_assets/kmodules/lisa/Makefile

```diff
@@ -25,23 +25,36 @@ else
 	KERNEL_SRC ?= /lib/modules/`uname -r`/build
 endif
 
-VMLINUX_H = $(MODULE_OBJ)/vmlinux.h
-SYMBOL_NAMESPACES_H = $(MODULE_OBJ)/symbol_namespaces.h
-
 # kbuild part of makefile. Only Kbuild-related targets should be used here to
 # avoid any sort of clash.
 ifneq ($(KERNELRELEASE),)
 
 LISA_KMOD_NAME ?= lisa
 obj-m := $(LISA_KMOD_NAME).o
 $(LISA_KMOD_NAME)-y := main.o tp.o wq.o features.o pixel6.o
-ldflags-y += -T $(M)/features.lds
-
-clean-files := vmlinux.h
 
 # -fno-stack-protector is needed to possibly undefined __stack_chk_guard symbol
 ccflags-y = -I$(MODULE_SRC) -std=gnu11 -fno-stack-protector -Wno-declaration-after-statement
 
+FEATURES_LDS := features.lds
+
+# in-tree build
+ifneq ($(srctree),.)
+
+EXTRA_CFLAGS += -I$(srctree) -D_IN_TREE_BUILD
```

Comment:
`EXTRA_CFLAGS` is deprecated, use ccflags-y instead (make sure to append to and not discard the existing ccflags-y):

```
    ccflags-y, asflags-y and ldflags-y
	These three flags apply only to the kbuild makefile in which they
	are assigned. They are used for all the normal cc, as and ld
	invocations happening during a recursive build.
	Note: Flags with the same behaviour were previously named:
	EXTRA_CFLAGS, EXTRA_AFLAGS and EXTRA_LDFLAGS.
	They are still supported but their usage is deprecated.
```
https://www.kernel.org/doc/Documentation/kbuild/makefiles.txt

Reply:
Ah right, I copied that line from Google's tracing module, of course it's deprecated :))